README.rst

```diff
@@ -317,6 +317,126 @@ specified in ``$HOME/.teuthology.yaml``::
+instance http://entercloudsuite.com/). If the cinder API is not
+available (for instance https://www.ovh.com/fr/cloud/), some jobs
+won't run because they expect volumes attached to each instance.
+
+Setup OpenStack at Enter Cloud Suite
+------------------------------------
+
+* create an account and `login the dashboard <https://dashboard.entercloudsuite.com/>`_
+* `create an Ubuntu 14.04 instance
+  <https://dashboard.entercloudsuite.com/console/index#/launch-instance>`_
+  with 1GB RAM and a public IP and destroy it immediately afterwards.
+* get $HOME/openrc.sh from `the horizon dashboard <https://horizon.entercloudsuite.com/project/access_and_security/?tab=access_security_tabs__api_access_tab>`_
+
+The creation/destruction of an instance via the dashboard is the
+shortest path to create the network, subnet and router that would
+otherwise need to be created via the neutron API.
+
+Setup OpenStack at OVH
+----------------------
+
+It is cheaper than EnterCloudSuite but does not provide volumes (as
+of August 2015) and is therefore unfit to run teuthology tests that
+require disks attached to the instance. Each instance has a public IP
+by default.
+
+* `create an account <https://www.ovh.com/fr/support/new_nic.xml>`_
+* get $HOME/openrc.sh from `the horizon dashboard <https://horizon.cloud.ovh.net/project/access_and_security/?tab=access_security_tabs__api_access_tab>`_
+
+Setup
+-----
+
+* Get and configure teuthology::
+
+    $ git clone -b wip-6502-openstack-v3 http://github.com/dachary/teuthology
+    $ cd teuthology ; ./bootstrap install
+    $ source virtualenv/bin/activate
+
+Get OpenStack credentials and test it
+-------------------------------------
+
+* follow the `OpenStack API Quick Start <http://docs.openstack.org/api/quick-start/content/index.html#cli-intro>`_
+* source $HOME/openrc.sh
+* verify the OpenStack client works::
+
+    $ openstack server list
+    +----+------------+--------+------------+-------------+-------------------------+
+    | ID | Name       | Status | Task State | Power State | Networks                |
+    +----+------------+--------+------------+-------------+-------------------------+
+    +----+------------+--------+------------+-------------+-------------------------+
+* upload your ssh public key with::
+
+    $ openstack keypair create --public-key ~/.ssh/id_rsa.pub myself
+    +-------------+-------------------------------------------------+
+    | Field       | Value                                           |
+    +-------------+-------------------------------------------------+
+    | fingerprint | e0:a3:ab:5f:01:54:5c:1d:19:40:d9:62:b4:b3:a1:0b |
+    | name        | myself                                          |
+    | user_id     | 5cf9fa21b2e9406b9c4108c42aec6262                |
+    +-------------+-------------------------------------------------+
+
+Usage
+-----
+
+* Run the dummy suite as a test::
+
+    $ teuthology-openstack --key-name myself --suite dummy
```

Comment:
@zmc I'm guessing the troubles you have with authentication via paramiko are related to paramiko/paramiko#387

A simple test is http://paste2.org/ZagzFYZX

Maybe orchestra would work but I would have to add support for the key_filename option and figure out how to catch the exceptions properly.